client/karma.js

```diff
@@ -234,18 +230,15 @@ function Karma (socket, iframe, opener, navigator, location, document) {
       socket.emit('result', resultsBuffer)
       resultsBuffer = []
     }
-
-    if (self.config.clearContext) {
-      // A test could have incorrectly issued a navigate. To clear the context
-      // we will navigate the iframe. Delay ours to ensure the error from an
-      // incorrect navigate is processed.
-      setTimeout(clearContext)
-    }
-
-    socket.emit('complete', result || {}, function () {
+    // A test could have incorrectly issued a navigate. Wait one turn
+    // to ensure the error from an incorrect navigate is processed.
+    setTimeout(() => {
       if (returnUrl) {
-        location.href = returnUrl
+        navigateContextTo(returnUrl)
```

Comment:
Actually on the second thought, I'm not sure if this is the correct change. From the [commit](https://github.com/karma-runner/karma/commit/6af2c897f3b35060a146efdef7da597ba53d8cdd) it sounds like the intention was to redirect the client, not context.

Do you know what is the use case for this feature? It was added a long time ago and I can't find any information as for why it is needed. Maybe we should consider removing this feature altogether and maybe restore it post v6 if anybody complains and can present a feasible use case?

Reply:
I agree that this feature we don't understand maybe should be deleted. What I did was simplify the implementation. Now the navigation occurs as soon as we send the `complete` event to the server.  It's possible that this will break someone, but then we will know who to ask "what is this for" and fix as appropriate. If not, this implementation is a lot less mysterious than the one before.  It basically just navigates the client if the url param is set, full stop.

Reply:
Yeah, I understand that. The comment was specifically about the `returnUrl`, which previously redirected the client (i.e. whole browser window) and was changed to redirect context (i.e. where tests are run). LGTM with the amended change.